sycl/source/detail/graph_impl.cpp

```diff
@@ -0,0 +1,417 @@
+//==--------- graph_impl.cpp - SYCL graph extension -----------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <detail/graph_impl.hpp>
+#include <detail/handler_impl.hpp>
+#include <detail/kernel_arg_mask.hpp>
+#include <detail/program_manager/program_manager.hpp>
+#include <detail/queue_impl.hpp>
+#include <detail/scheduler/commands.hpp>
+#include <sycl/feature_test.hpp>
+#include <sycl/queue.hpp>
+
+// Developer switch to use emulation mode on all backends, even those that
+// report native support, this is useful for debugging.
+#define FORCE_EMULATION_MODE 1
```

Comment:
Shouldn't we submit the code with value 0? I suppose it's useful for debugging, but we lose native support performance benefits. Right?

Reply:
I believe this is due to the fact that we're splitting our work up into separate PRs to ease review. In our [PR with the complete changes](https://github.com/intel/llvm/pull/9375 ) this macro is set to zero and we do use backend command-buffers. So our intention is to get that behaviour merged. However, that PI/UR extension code with backend implementation will be its own follow-on split PR, which would flip this macro to 1.

Reply:
Yes that was my intention. I see two options: Either take it out for now because the switch doesn't do anything with this patch, or keep the flag and set it just for our initial PR and unset it in a follow up PR once the backend support is added.

Reply:
> take it out for now because the switch doesn't do anything with this patch

This sounds reasonable to me.

Reply:
Addressed in reble#212